pkg/services/serviceaccounts/api/token.go

```diff
@@ -160,6 +160,14 @@ func (api *ServiceAccountsAPI) CreateToken(c *models.ReqContext) response.Respon
 		}
 	}
 
+	if api.cfg.TokenExpirationDayLimit > 0 {
```

Comment:
```suggestion
	if api.cfg.TokenExpirationDayLimit {
```
I find that the code would be a bit more readable when we do not define the variable is -1 by default.

Given my comment on the option field to be unset by default and only set it when needed.

When need to then check that the tokenexpirationdaylimit is set to a value above zero.
But it might be worth looking into

Reply:
Regardless of having a default value or not, if we remove the conditional from here, an incorrect (but valid) value could be a negative value. This will produce unintended consequences.

Reply:
I was not clear in the text perhaps

> When need to then check that the tokenexpirationdaylimit is set to a value above zero.

 we do need to check that this setting is bigger than zero if this were to be implemented, but i do feel like that would makes sense to implement
 
wdyt?